tests/p2p/test_get_best_blockchain.py

```diff
@@ -257,6 +262,60 @@ def test_node_without_get_best_blockchain_capability(self):
         self.simulator.run(60)
         self.assertTrue(conn12.tr2.disconnecting)
 
+    @inlineCallbacks
+    def test_best_blockchain_within_connected_peers(self):
+        manager1 = self.create_peer()
+        manager2 = self.create_peer()
+        conn12 = FakeConnection(manager1, manager2, latency=0.05)
+        self.simulator.add_connection(conn12)
+        self.simulator.run(60)
+
+        # check /status before generate blocks
+        self.web = StubSite(StatusResource(manager1))
+        response = yield self.web.get("status")
+        data = response.json_value()
+        connections = data.get('connections')
+        self.assertEqual(len(connections['connected_peers']), 1)
+
+        # assert there is the genesis block
+        best_blockchain = connections['connected_peers'][0]['best_blockchain']
+        self.assertEqual(len(best_blockchain), 1)
+        # assert the block_info height is from genesis
+        raw_block_info_height = best_blockchain[0][1]
+        self.assertEqual(raw_block_info_height, 0)
+
+        # mine 20 blocks
+        miner = self.simulator.create_miner(manager1, hashpower=1e6)
+        miner.start()
+        trigger = StopAfterNMinedBlocks(miner, quantity=20)
+        self.assertTrue(self.simulator.run(1440, trigger=trigger))
+        miner.stop()
+        # let the blocks to propagate
+        self.simulator.run(60)
+
+        # check /status after mine blocks
+        response = yield self.web.get("status")
+        data = response.json_value()
+        connections = data.get('connections')
+        self.assertEqual(len(connections['connected_peers']), 1)
+
+        # assert default best_blockchain length
+        best_blockchain = connections['connected_peers'][0]['best_blockchain']
+        self.assertEqual(len(best_blockchain), 10)
+
+        # assert a raw_block_info can be converted to BlockInfo
+        try:
+            raw_block_info = best_blockchain[0]
+            block_info = BlockInfo.from_raw(raw_block_info)
+            # assert the first element height is from the lastest block mined
+            self.assertEqual(block_info.height, 20)
+        except ValueError:
+            self.fail('Block info not valid')
+
+        # assert decreasing order for a sequence of heights
+        height_sequence = [b[1] for b in best_blockchain]
+        self.assertTrue(test_list.test_decreasing_monotonicity(height_sequence))
```

Comment:
I'm not sure whether we should generalize this `test_decreasing_monotonicity()` method. It seems we can easily iterate and generate a more friendly error message.

Reply:
I removed the generalization, but brought the check to the test module as util function.